code/datums/elements/connect_loc.dm

```diff
@@ -0,0 +1,77 @@
+/// This element hooks a signal onto the loc the current object is on.
+/// When the object moves, it will unhook the signal and rehook it to the new object.
+/datum/element/connect_loc
+	element_flags = ELEMENT_BESPOKE | ELEMENT_DETACH
+	id_arg_index = 2
+
+	/// An assoc list of signal -> procpath to register to the loc this object is on.
+	var/list/connections
+
+	/// An assoc list of locs that are being occupied and a list of targets that occupy them.
+	var/list/targets = list()
```

Comment:
Surely there's a better way to do this, this'll get slow quite quickly

Reply:
There's not really, what about this do you think is slow?

Reply:
it's an assoc list, so anything you do to it will get slower the longer it is, which limits the use of the element, right?

Reply:
This list is only ever read in one place, which is ChangeTurf. It's mutated in some other places, but not *commonly*, unless it's something that is moving around a lot, I suppose. There's no other way to do this, no.

Reply:
@LemonInTheDark the comment left by oranges has our discussion about this.